cpp/src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITValue.cpp

```diff
@@ -541,8 +541,9 @@ JITValuePointer LLVMJITValue::castJITValuePrimitiveType(JITTypeTag target_jit_ta
   llvm::Value* source_lv = load();
   llvm::Value* target_lv = nullptr;
   if (source_type->isIntegerTy() && target_type->isIntegerTy()) {
-    target_lv =
-        getFunctionBuilder(parent_function_).CreateIntCast(source_lv, target_type, true);
+    target_lv = getFunctionBuilder(parent_function_)
+                    .CreateIntCast(
+                        source_lv, target_type, source_type->getScalarSizeInBits() != 1);
```

Comment:
Q: Here we should get whether `source_lv` is signed or not, right? If so, does `source_type->getScalarSizeInBits() != 1` mean that?

Reply:
see difference between 'zext .. to' and 'sext .. to' 
https://releases.llvm.org/2.6/docs/LangRef.html#convertops